wayland-client/src/event_queue.rs

```diff
@@ -615,21 +705,35 @@ fn queue_callback<
     Ok(())
 }
 
-type ObjectDataFactory = dyn Fn(&Message<ObjectId>) -> Option<Arc<dyn ObjectData>> + Send + Sync;
+fn early_callback<
+    I: Proxy + 'static,
+    U: Send + Sync + 'static,
+    State: Dispatch<I, U, State> + 'static,
+>(
+    handle: &Connection,
+    msg: Message<ObjectId>,
+    data: &mut State,
+    _: Arc<dyn ObjectData>,
+    qhandle: &QueueHandle<State>,
+) -> Result<(), DispatchError> {
+    let (proxy, event) = I::parse_event(handle, msg)?;
+    if let Some(proxy_data) = proxy.data() {
+        <State as Dispatch<I, U, State>>::event(data, &proxy, event, &proxy_data, handle, qhandle);
```

Comment:
Do we need to handle event_created_child here?

Reply:
No, that's already handled (and is one of the reasons why `make_early_data` needs the type of the userdata even if it doesn't use it directly).